include/aspect/material_model/interface.h

```diff
@@ -515,9 +515,63 @@ namespace aspect
         {}
     };
 
+    template<int dim>
+    class NamedAdditionalMaterialOutputs: public AdditionalMaterialOutputs<dim>
+    {
+      public:
+        NamedAdditionalMaterialOutputs(const std::vector<std::string> output_names)
+          :
+          names(output_names)
+        {}
```

Comment:
can you move the implementation of this and the other functions into the .cc file?

I recognize that that is also missing from the previous class -- working on that.